### DISCUSSION TITLE
Knowing when Menu has been opened

### DISCUSSION THREAD

**Discussion by haf** [tailwindlabs/headlessui#776] [Help]
Hello,

What's the right/a great way to know if Menu has been opened and its children rendered? OnClick handler on the `Menu.Button`?

Thanks,
Henrik

**Comment by dospunk** [comment]
You should use the `open` slot prop provided to `Menu`, as shown in the second example under [this section](https://headlessui.dev/react/menu#showing-hiding-the-menu) in the docs

**Reply by haf** [reply]
Yes, I saw that, but I'm looking for a callback or a `prop`, not a render prop, because I can't access that from the menu's parent component. I simply need to dynamically do a request when the menu is opened.
